prototype-rest-model/src/main/java/org/jboss/pnc/rest/model/BuildConfigurationRef.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.rest.model;
+
+import org.jboss.pnc.rest.model.enums.BuildType;
+
+import java.time.Instant;
+import java.util.Date;
+import java.util.Map;
+
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class BuildConfigurationRef {
+
+    private final int id;
+
+    private final String name;
+
+    private final String description;
+
+    private final String buildScript;
+
+    private final String scmRevision;
+
+    private final Instant creationTime;
+
+    private final Instant lastModificationTime;
+
+    private final boolean archived;
+
+    private final BuildType buildType;
+
+    private final Integer productVersionId;
+
+    private final Map<String, String> genericParameters;
+
+    @JsonPOJOBuilder(withPrefix = "")
+    public static final class BuildConfigurationRefBuilder {
```

Comment:
I'm purely splitting hairs for the sake of an argument here ;) ........ but why not just call the static inner class Builder...

It's more concise and readable to have:

var bld = new BuildConfigurationRef.Builder();

vs 

var bld = new BuildConfigurationRef.BuildConfigurationRefBuilder();

Reply:
Yeah, I'm not particularly bothered, was just interested.

Reply:
We'll use `BuildConfigurationRef.Builder` version